enclave_manager/avalon_enclave_manager/singleton/enclave_manager.py

```diff
@@ -247,131 +189,222 @@ def process_work_order_by_id(self, kv_helper, wo_id):
-            status of the work order receipt and updater signature update in
-            the receipt.
+        Submits workorder request to Worker enclave and retrieves the response
         """
-        receipt_entry = kv_helper.get("wo-receipts", wo_id)
-        if receipt_entry:
-            update_type = None
-            if "error" in wo_json_resp and \
-                    wo_json_resp["error"]["code"] != \
-                    WorkOrderStatus.PENDING.value:
-                update_type = ReceiptCreateStatus.FAILED.value
-            else:
-                update_type = ReceiptCreateStatus.PROCESSED.value
-            receipt_obj = WorkOrderReceiptRequest()
-            wo_receipt = receipt_obj.update_receipt(
-                wo_id,
-                update_type,
-                wo_json_resp,
-                self.private_key
-            )
-            updated_receipt = None
-            # load previous updates to receipt
-            updates_to_receipt = kv_helper.get("wo-receipt-updates", wo_id)
-            # If it is first update to receipt
-            if updates_to_receipt is None:
-                updated_receipt = []
-            else:
-                updated_receipt = json.loads(updates_to_receipt)
-                # Get the last update to receipt
-                last_receipt = updated_receipt[len(updated_receipt) - 1]
-
-                # If receipt updateType is completed,
-                # then no further update allowed
-                if last_receipt["updateType"] == \
-                        ReceiptCreateStatus.COMPLETED.value:
-                    logger.info(
-                        "Receipt for the workorder id %s is completed " +
-                        "and no further updates are allowed",
-                        wo_id)
-                    return
-            updated_receipt.append(wo_receipt)
-
-            # Since receipts_json is jrpc request updating only params object.
-            kv_helper.set("wo-receipt-updates", wo_id, json.dumps(
-                updated_receipt))
-            logger.info("Receipt for the workorder id %s is updated to %s",
-                        wo_id, wo_receipt)
+        try:
+            wo_request = work_order_request.SgxWorkOrderRequest(
+                self.enclave_data,
+                input_json_str)
+            wo_response = wo_request.execute()
+
+            try:
+                json_response = json.dumps(wo_response, indent=indent)
+            except Exception as err:
+                logger.error("ERROR: Failed to serialize JSON; %s", str(err))
+                wo_response["Response"]["Status"] = WorkOrderStatus.FAILED
+                wo_response["Response"]["Message"] = "Failed to serialize JSON"
+                json_response = json.dumps(wo_response)
+
+        except Exception as e:
+            logger.error("failed to execute work order; %s", str(e))
+            wo_response["Response"]["Status"] = WorkOrderStatus.FAILED
```

Comment:
As per spec, error response should have the following structure.
{
    "jsonrpc": "2.0",    // as per JSON RPC spec
    "id": <integer>,    // the same as in input
    "error": {      // as per JSON RPC spec
        "code": <integer>,
        "message": <string>,
        "data": <implementation specific data>
    }
}

Reply:
Shall be taken care in separate PR - https://github.com/hyperledger/avalon/issues/441

Reply:
Ok